gridData/OpenDX.py

```diff
@@ -678,8 +679,13 @@ def parse(self,DXfield):
         self.currentobject = None           # containers for data
         self.objects = []                   # |
         self.tokens = []                    # token buffer
-        with open(self.filename, 'r') as self.dxfile:
-            self.use_parser('general')      # parse the whole file and populate self.objects
+
```

Comment:
Do something like
```python
if filename.endswith('.gz'):
   with gzip.open(self.filename, 'rt') as self.dxfile:
                self.use_parser('general')
else:
    with open(self.filename, 'r') as self.dxfile:
                self.use_parser('general')      # parse the whole file and populate self.objects
```

Once we add `openany()` to the code, this can be replaced with
```python
with openany(filename):
     self.use_parser('general')
```